### ISSUE TITLE
no-extra-parens new option to avoid conflicts with no-sequences

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#11916]
**What rule do you want to change?**

no-extra-parens

**Does this change cause the rule to produce more or fewer warnings?**

fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

new option

**Please provide some example code that this change will affect:**

```js
/* eslint  no-extra-parens: "error" */
/* eslint  no-sequences: "error" */

(a, b); // 'no-extra-parens' warning

if ((a, b)); // 'no-extra-parens' warning

while((a, b)); // 'no-extra-parens' warning
```

**What does the rule currently do for this code?**

Fixes it to:

```js
/* eslint  no-extra-parens: "error" */
/* eslint  no-sequences: "error" */

a, b; // 'no-sequences' warning

if (a, b); // 'no-sequences' warning

while(a, b); // 'no-sequences' warning
```

**What will the rule do after it's changed?**

```js
/* eslint  no-extra-parens: ["error", "all", { "sequenceExpressions": false }] */
/* eslint  no-sequences: "error" */

(a, b); // no warnings

if ((a, b)); // no warnings

while((a, b)); // no warnings
```

**Are you willing to submit a pull request to implement this change?**

Yes

**Comment by platinumazure** [comment]
@eslint/eslint-team This just needs a champion and we could accept this!

**Comment by kaicataldo** [comment]
I'll champion this.

**Comment by mdjermanovic** [comment]
I'm working on this :)

What should be the name of the new option? The last was `enforceForArrowConditionals`

Maybe `enforceForSequenceExpressions` instead of just `sequenceExpressions`?

**Comment by platinumazure** [comment]
I like `enforceForSequenceExpressions` personally. I generally don't like option names without a verb indicating what will happen if you turn it on. (Just my two cents; other team members may have their own opinion.)
